dms/models/directory.py

```diff
@@ -194,6 +195,40 @@ class DmsDirectory(models.Model):
                 """,
     )
 
+    def check_access_rule(self, operation):
+        for record in self.filtered(
+            lambda x: (
+                (not x.is_root_directory and x.storage_id.save_type == "attachment")
+                or (x.is_root_directory and x.root_storage_id.save_type == "attachment")
+            )
+            and x.res_model
+        ):
+            model = record.res_model
+            try:
+                if record.res_id:
+                    self.env[model].browse(record.res_id).check_access_rule(operation)
+                else:
+                    self.env[model].check_access_rights(operation)
+            except AccessError:
+                raise AccessError(
+                    _("No tienes permisos de %s para acceder al registro %s,%s")
+                    % (operation, record.res_model, record.res_id)
+                )
```

Comment:
This can be much simpler. Besides, `check_access_rights` must be called always:

```suggestion
            # Check access to inherited model (and record)
            model = self.env[record.res_model]
            model.check_access_rights(operation)
            if record.res_id:
                model.browse(record.res_id).check_access_rule(operation)
```